tests/misc/rbtree/src/main.c

```diff
@@ -0,0 +1,211 @@
+/*
+ * Copyright (c) 2018 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <zephyr.h>
+#include <tc_util.h>
+#include <ztest.h>
+#include <misc/rb.h>
+
+#define CHECK(n) \
+	zassert_true(!!(n), "Tree check failed: [ " #n " ] @%d", __LINE__)
+
+#define MAX_NODES 2048
+
+static struct rbtree tree;
+
+static struct rbnode nodes[MAX_NODES];
+
+/* Bit is set if node is in the tree */
+static unsigned int node_mask[(MAX_NODES + 31)/32];
+
+/* Array of nodes dumed via rb_walk */
+static struct rbnode *walked_nodes[MAX_NODES];
+
+void set_node_mask(int node, int val)
+{
+	unsigned int *p = &node_mask[node / 32];
+	unsigned int bit = 1 << (node % 32);
```

Comment:
This should be `1u << (node % 32)`: shifting left by 31 on an `int` is undefined behavior